src/node/node.cc

```diff
@@ -387,28 +388,57 @@ void RaftNode::RescheduleElection() {
     timer_manager_->ResetTimer(RaftcppConstants::TIMER_ELECTION, GetRandomizedElectionTimeout());
 }
 
-void RaftNode::ReplicateOneRound(int64_t node_id) {
+void RaftNode::ReplicateOneRound(int64_t node_id, std::shared_ptr<int64_t> quorum) {
     AppendEntriesRequest request;
     auto context = std::make_shared<grpc::ClientContext>();
     auto response = std::make_shared<AppendEntriesResponse>();
     all_rpc_clients_[node_id]->async()
         ->HandleRequestAppendEntries(context.get(), &request, response.get(),
-        [context, response](grpc::Status s){
-        // TODO asynchronous replication
+        [context, response, quorum, this](grpc::Status s){
+            if(!s.ok()) {
+                RAFTCPP_LOG(RLL_DEBUG) << s.error_code() << ": " << s.error_message();
+                return;
+            }
+            std::lock_guard<std::recursive_mutex> guard{mutex_};
+            if (response->term() < curr_term_ || curr_state_ != RaftState::LEADER) {
+                return;
+            }
+            if (response->term() > curr_term_) {
+                BecomeFollower(response->term(), response->leader_id());
+                return;
+            }
+            
+            // TODO
+
+            if (quorum) {
```

Comment:
It's better to add `UpdateQuorum()` method instead of passing shared_ptr for updating purpose, like:
```C++
void RaftNode::ReplicateOneRound(int64_t node_id, std::shared_ptr<int64_t> quorum) {
    AppendEntriesRequest request;
    auto context = std::make_shared<grpc::ClientContext>();
    auto response = std::make_shared<AppendEntriesResponse>();
    all_rpc_clients_[node_id]->async()
        ->HandleRequestAppendEntries(context.get(), &request, response.get(),
        [context, response, quorum, this](grpc::Status s){
                   // xxxx
                    this->UpdateQuorum(); // or this->IncreseQuorum();
                   // xxx
           };